src/integrations/blocks/abstract-dynamic-block-v3.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Yoast\WP\SEO\Integrations\Blocks;
+
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+
+/**
+ * Dynamic_Block class.
+ */
+abstract class Dynamic_Block_V3 implements Integration_Interface {
+
+	/**
+	 * The name of the block.
+	 *
+	 * @var string
+	 */
+	protected $block_name;
+
+	/**
+	 * The editor script for the block.
+	 *
+	 * @var string
+	 */
+	protected $script;
+
+	/**
+	 * The base path for the block.
+	 *
+	 * @var string
+	 */
+	protected $base_path;
+
+	/**
+	 *  Returns the conditionals based on which this loadable should be active.
+	 *
+	 * @return array<string>
+	 */
+	public static function get_conditionals() {
+		return [];
+	}
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * This is the place to register hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'init', [ $this, 'register_block' ], 11 );
+	}
+
+	/**
+	 * Registers the block.
+	 *
+	 * @return void
+	 */
+	public function register_block() {
+		\register_block_type(
+			$this->base_path . $this->block_name . '/block.json',
+			[
+				'editor_script'   => $this->script,
+				'render_callback' => [ $this, 'present' ],
```

Comment:
I was wondering if we could use the `render` in `block.json` to achieve this: https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#render

It would bring all the definitions to the `block.json`.
But, I'm not sure if that is nicer, since now you have the object oriented approach here.